File: web/src/layout/ComponentTextXml.ts
//
// ComponentTextXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './component_text.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ComponentTextXml
export class ComponentTextXml {
    xmlRoot!: HTMLElement;
    label!: HTMLElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.label = getViewById<HTMLElement>(view, "label");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/ComponentTextXml.kt
//
// ComponentTextXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class ComponentTextXml {

    lateinit var label: TextView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.component_text, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        label = view.findViewById<TextView>(R.id.label)
        
        
        
        return view
    }
}